clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1408,6 +1413,11 @@ void handleFailure(Throwable throwable) {
     @Override
     public DeleteTopicsResult deleteTopics(Collection<String> topicNames,
                                            DeleteTopicsOptions options) {
+        // avoid unnecessary metadata lookup
+        if (topicNames.isEmpty()) {
```

Comment:
Below we check and construct a subset of `validTopicNames`; we should skip if that subset is empty, and just return the future with exception with the rest invalid topic name.s